scarb/src/ops/fmt.rs

```diff
@@ -42,28 +45,97 @@ pub fn format(opts: FmtOptions, ws: &Workspace<'_>) -> Result<bool> {
 
     let all_correct = AtomicBool::new(true);
 
+    if let Some(target_path) = &opts.target_path {
+        if target_path.is_file() {
+            format_single_file(target_path, &opts, ws, &all_correct)?;
+            return Ok(all_correct.load(Ordering::Acquire));
+        }
+        let pkg = ws
+            .members()
+            .find(|member| target_path.starts_with(member.root()));
+
+        if let Some(pkg) = pkg {
+            format_package(&pkg, &opts, ws, &all_correct)?;
+            return Ok(all_correct.load(Ordering::Acquire));
+        } else {
+            return Err(anyhow::anyhow!(
+                "File {:?} is not part of the workspace.",
```

Comment:
```suggestion
                "path {:?} is not part of the workspace",
```

Reply:
There is a shorthand  for`return Err(anyhow!` called `anyhow::bail` :)

Reply:
done